xbmc/addons/kodi-addon-dev-kit/cmake/test/abi-interface-test.cmake

```diff
@@ -0,0 +1,181 @@
+option(DEVKIT_TEST "To test dev-kit" TRUE)
+option(DEVKIT_TEST_CPP "To test dev-kit headers correct in \"C++\" compile" FALSE)
+option(DEVKIT_TEST_STOP_ON_ERROR "To stop build if during \"C\" ABI test an error was found" FALSE)
+
+if(NOT DEVKIT_TEST)
+  return()
+endif()
+
+include(CheckCSourceCompiles)
+include(CheckCXXSourceCompiles)
+
+list(APPEND CMAKE_REQUIRED_INCLUDES ${CMAKE_CURRENT_SOURCE_DIR}/include/kodi
+                                    ${CMAKE_CURRENT_SOURCE_DIR}/../..)
+
+message(STATUS "################################################################################")
+message(STATUS "# Run test about dev-kit headers for correct style")
+message(STATUS "#")
+
+# Set minimal required defines
+if(CORE_SYSTEM_NAME STREQUAL android)
+  set(DEVKIT_TEST_DEFINES "\
+#define HAS_GLES 3
+#define TARGET_POSIX
+#define TARGET_LINUX
+#define TARGET_ANDROID
+")
+elseif(CORE_SYSTEM_NAME STREQUAL darwin_embedded)
+  set(DEVKIT_TEST_DEFINES "\
+#define HAS_GLES 2
+#define TARGET_POSIX
+#define TARGET_DARWIN
+#define TARGET_DARWIN_EMBEDDED
+")
+elseif(CORE_SYSTEM_NAME STREQUAL osx)
+  set(DEVKIT_TEST_DEFINES "\
+#define HAS_GL 1
+#define TARGET_DARWIN
+#define TARGET_DARWIN_OSX
+")
+elseif(CORE_SYSTEM_NAME STREQUAL windows)
+  set(DEVKIT_TEST_DEFINES "\
+#define HAS_DX 1
+#define WIN32
+#define TARGET_WINDOWS
+#define TARGET_WINDOWS_DESKTOP
+")
+elseif(CORE_SYSTEM_NAME STREQUAL linux)
+  set(DEVKIT_TEST_DEFINES "\
+#define HAS_GL 1
+#define TARGET_POSIX
+#define TARGET_LINUX
+")
+else()
+  set(DEVKIT_TEST_DEFINES "\
+#define HAS_GL 1
+#define TARGET_POSIX
+#define TARGET_FREEBSD
+")
+endif()
```

Comment:
all those defines shouldn't be necessary if this test was run after ArchSetup. ARCH_DEFINES and SYSTEM_DEFINES should be enough.
In fact I would move add_subdirectory(xbmc/addons/kodi-addon-dev-kit) way down after the kodi build.

Reply:
Knows what you mean, just want to leave it independent of Kodi for the time being to be able to call it up separately (also primarily to quickly do it in between when editing the header).

In addition, cmake things used there should not be dependent on Kodi itself and should be able to act completely independently. So that this folder can also be used outside of Kodi.

Will be expanded in the future and these positions revised.

-------------------

He is also temporary in Kodi's CMakeLists.txt and wants to leave it out for the time being. At the end when it is there (and all headers are correct), I imagine to bring it there and in Jenkins a **"`-DDEVKIT_TEST=1 -DDEVKIT_TEST_CPP=1 -DDEVKIT_TEST_STOP_ON_ERROR=1`"** to complain about errors.

But with normal creation (by users) all three as **FALSE**, because it is not of direct interest there.

-------------------

Even if I have cleaned up the addon header of everyone, create a new github project here in xbmc.

Is to find together with team a way to use this folder completely independently and to bypass the TM problem with binary addon for sources that use Kodi under other names